core/SizeSystem.js

```diff
@@ -0,0 +1,259 @@
+        if (size.renderSizeChanged) renderSizeChanged(node, components, size);
+        if (size.fromComponents(components)) sizeChanged(node, components, size);
+    }
+};
+
+// private methods
+
+/**
+ * Private method to alert the node and components that size mode changed.
+ *
+ * @method
+ * @private
+ *
+ * @param {Node} node Node to potentially call sizeModeChanged on
+ * @param {Array} components a list of the nodes' components
+ * @param {Size} the size component
+ */
+function sizeModeChanged (node, components, size) {
+    var sizeMode = size.getSizeMode();
+    var x = sizeMode[0];
+    var y = sizeMode[1];
+    var z = sizeMode[2];
+    if (node.onSizeModeChange) node.onSizeModeChange(x, y, z);
+    for (var i = 0, len = components.length ; i < len ; i++)
+        if (components[i] && components[i].onSizeModeChange)
+            components[i].onSizeModeChange(x, y, z);
+    size.sizeModeChanged = false;
+}
+
+/**
+ * Private method to alert the node and components that absoluteSize changed.
+ *
+ * @method
+ * @private
+ *
+ * @param {Node} node Node to potentially call onAbsoluteSizeChange on
+ * @param {Array} components a list of the nodes' components
+ * @param {Size} the size component
+ */   
+function absoluteSizeChanged (node, components, size) {
+    var absoluteSize = size.getAbsoluteSize();
+    var x = absoluteSize[0];
+    var y = absoluteSize[1];
+    var z = absoluteSize[2];
+    if (node.onAbsoluteSizeChange) node.onAbsoluteSizeChange(x, y, z);
+    for (var i = 0, len = components.length ; i < len ; i++)
+        if (components[i] && components[i].onAbsoluteSizeChange)
+            components[i].onAbsoluteSizeChange(x, y, z);
+    size.absoluteSizeChanged = false;
+}
+
+/**
+ * Private method to alert the node and components that the proportional size changed.
+ *
+ * @method
+ * @private
+ *
+ * @param {Node} node Node to potentially call onProportionalSizeChange on
+ * @param {Array} components a list of the nodes' components
+ * @param {Size} the size component
+ */
+function proportionalSizeChanged (node, components, size) {
+    var proportionalSize = size.getProportionalSize();
+    var x = proportionalSize[0];
+    var y = proportionalSize[1];
+    var z = proportionalSize[2];
+    if (node.onProportionalSizeChange) node.onProportionalSizeChange(x, y, z);
+    for (var i = 0, len = components.length ; i < len ; i++)
+        if (components[i] && components[i].onProportionalSizeChange)
+            components[i].onProportionalSizeChange(x, y, z);
+    size.proportionalSizeChanged = false;
+}
+
+/**
+ * Private method to alert the node and components that differential size changed.
+ *
+ * @method
+ * @private
+ *
+ * @param {Node} node Node to potentially call onDifferentialSize on
+ * @param {Array} components a list of the nodes' components
+ * @param {Size} the size component
+ */
+function differentialSizeChanged (node, components, size) {
+    var differentialSize = size.getDifferentialSize();
```

Comment:
@DnMllr `size` doesn't seem to have a `getDifferentialSize()` method. Using`node` and changing this to `var differentialSize = node.getDifferentialSize();` seems to work well though.